drivers/storage/azure/utils/utils.go

```diff
@@ -0,0 +1,77 @@
+// +build !libstorage_storage_driver libstorage_storage_driver_azure
+
+package utils
+
+import (
+	"bufio"
+	"bytes"
+	"os"
+
+	"github.com/akutz/goof"
+
+	"github.com/codedellemc/libstorage/api/types"
+	"github.com/codedellemc/libstorage/drivers/storage/azure"
+)
+
+func checkAzureMarkInFile(ctx types.Context) bool {
+	file := "/var/lib/dhcp/dhclient.eth0.leases"
+	pattern := []byte("unknown-245")
+
+	f, err := os.Open(file)
+	if err != nil {
+		ctx.Debug("Specific file (" + file + ") could not be opened:")
+		ctx.Debug(err)
+		return false
+	}
+	defer f.Close()
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		if bytes.Contains(scanner.Bytes(), pattern) {
+			return true
+		}
+	}
+	if err := scanner.Err(); err != nil {
+		ctx.Debug("Specific file (/var/lib/dhcp/dhclient.eth0.leases) could not be read:")
+		ctx.Debug(err)
+	}
+	return false
+}
+
+// IsAzureInstance returns a flag indicating whether the executing host is an Azure
+// instance based on whether or not the metadata URL can be accessed.
+func IsAzureInstance(ctx types.Context) (bool, error) {
+	// http://blog.mszcool.com/index.php/2015/04/detecting-if-a-virtual-machine-runs-in-microsoft-azure-linux-windows-to-protect-your-software-when-distributed-via-the-azure-marketplace/
+	if id := os.Getenv("AZURE_INSTANCE_ID"); id != "" {
+		return true, nil
+	}
+	result := checkAzureMarkInFile(ctx)
+	return result, nil
+}
+
+// InstanceID returns the instance ID for the local host.
+func InstanceID(ctx types.Context) (*types.InstanceID, error) {
+	hostname := os.Getenv("AZURE_INSTANCE_ID")
+	if hostname == "" {
+		isAzure, err := IsAzureInstance(ctx)
+		if err != nil {
+			return nil, err
+		}
+		if !isAzure {
+			return nil, goof.New("Executing outside of Instance. InstanceID could not be obtained.")
+		}
+
+		// UUID can be obtained as descried in https://azure.microsoft.com/en-us/blog/accessing-and-using-azure-vm-unique-id/
+		// but this code will use hostname as ID
+
+		hostname, err = os.Hostname()
```

Comment:
Azure doesn't set AZURE_INSTANCE_ID inside VM-s. This variable was implemented for unit tests only.
Azure set hostname inside of VM equals to name of VM in Azure engine. And this name is unique in the engine.
So because there is no other way to obtain instance identifier (that is suitable for REST API) then we get it from the hostname.